src/plugins/intel_cpu/src/graph_optimizer.cpp

```diff
@@ -3273,24 +3286,34 @@ void GraphOptimizer::DropRedundantMemoryOutput(Graph &graph) {
         graph.RemoveEdge(memoryOutputNode->getParentEdgeAt(0));
         // there are no output edges from MemoryOutput nodes
 
+        auto memInpNd = std::dynamic_pointer_cast<node::MemoryInput>(node);
```

Comment:
Next line can cover nullptr check,
`auto subGraph = memInpNd ? memInpNd->getSubGraph() : nullptr;`

Reply:
Actually no, as there may be node of any type, which can't be casted to `node::MemoryInput`, so that all the further actions are meaningless, therefore an exception must be throw.

Reply:
I am a bit confused here.
`auto subGraph = memInpNd ? memInpNd->getSubGraph() : nullptr;`
If `memInpNd`  is empty, subGraph will be nullptr. And `MemoryInputSingle` can accept empty subgraph. memInpNd also only use here.